core/src/main/scala/sttp/client3/internal/package.scala

```diff
@@ -30,6 +34,18 @@ package object internal {
     transfer()
   }
 
+  private[client3] def withCharset(headers: Map[String, String], encoding: String): Map[String, String] = {
```

Comment:
maybe we could add an `addCharsetToContentType(charset: String): Part` method on `Part`, that could nicely encapsulate that logic